### PR TITLE
Allow PHP 8.1

### PR DESCRIPTION
**Pull request by snapshotpl**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Allow install on PHP 8.1

Needs:

 * [x] https://github.com/bshaffer/oauth2-server-php/pull/1021 merged+released

### CODE REVIEW

**Comment by Ocramius** [comment]
Welp, `- Root composer.json requires laminas/laminas-coding-standard ~2.2.0 -> satisfiable by laminas/laminas-coding-standard[2.2.0, 2.2.1].`

Guess `laminas/laminas-coding-standard:~2.3` is needed here.

**Review by Ocramius — CHANGES_REQUESTED** [review]
Welp, `- Root composer.json requires laminas/laminas-coding-standard ~2.2.0 -> satisfiable by laminas/laminas-coding-standard[2.2.0, 2.2.1].`

Guess `laminas/laminas-coding-standard:~2.3` is needed here.

**Review by Ocramius — CHANGES_REQUESTED** [review]
```
 1) LaminasTest\ApiTools\OAuth2\Controller\AuthControllerTest::testResource
Laminas\View\Exception\DomainException: Laminas\View\Renderer\PhpRenderer::render: received View Model argument, but template is empty
```

Meanwhile, PHP 7.3 can go away :)

**Comment by snapshotpl** [comment]
Fix requested https://github.com/bshaffer/oauth2-server-php/pull/1021

**Review by Ocramius — CHANGES_REQUESTED** [review]
For the CS issues, could we perhaps switch to psr-11, if servicemanager is forced to be at the last version? 🤔

**Comment by snapshotpl** [comment]
CS issue fixed, so now waiting for https://github.com/bshaffer/oauth2-server-php/pull/1021

**Review by Ocramius — APPROVED** [review]

**Comment by snapshotpl** [comment]
After squash looks like GA stuck, but before all passed https://github.com/laminas-api-tools/api-tools-oauth2/actions/runs/2420613249 so just waiting for new release oauth2 by @bshaffer

File: composer.json
Comment (Ocramius) [inline composer.json]: To be adjusted before merge

File: src/Factory/PdoAdapterFactory.php
Comment (Ocramius) [inline src/Factory/PdoAdapterFactory.php]: This is still a BC break :thinking:

Comment (Ocramius) [reply src/Factory/PdoAdapterFactory.php]: (applies to other factories as well)

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Factory/AuthControllerFactory.php
Comment (Ocramius) [inline src/Factory/AuthControllerFactory.php:39]: @snapshotpl if we keep the BC breaks in, we need a major: can we keep the dead `public` methods for a while, and mark them `@deprecated` instead?

File: src/Factory/AuthControllerFactory.php
Comment (Ocramius) [inline src/Factory/AuthControllerFactory.php]: Can we somehow keep `FactoryInterface` to avoid a BC break? :thinking:

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by bshaffer** [comment]
> CS issue fixed, so now waiting for [bshaffer/oauth2-server-php#1021](https://github.com/bshaffer/oauth2-server-php/pull/1021)

Fixed in https://github.com/bshaffer/oauth2-server-php/releases/tag/v1.12.1

**Comment by snapshotpl** [comment]
Thanks @bshaffer! All tests are green now!

**Comment by snapshotpl** [comment]
Not sure why php-cs enforced `interop\container` lowercase :thinking: . But still works

**Comment by Ocramius** [comment]
> Not sure why php-cs enforced `interop\container` lowercase thinking . But still works

It will break autoloading (depending on autoloading order), so I think we need to either ignore or skip the rule for now :thinking:

File: src/Factory/PdoAdapterFactory.php
Comment (Ocramius) [inline src/Factory/PdoAdapterFactory.php]: These reported CS issues need to be suppressed: I think we may want to run `phpcs -s` to see **which** rule it is, that is raising them.

**Review by Ocramius — APPROVED** [review]

File: src/Factory/PdoAdapterFactory.php
Comment (Ocramius) [reply src/Factory/PdoAdapterFactory.php]: Once the build is green due to this, I'm happy to merge+release :)

Comment (snapshotpl) [reply src/Factory/PdoAdapterFactory.php]: `WebimpressCodingStandard.PHP.CorrectClassNameCase.Invalid` @michalbundyra

Comment (Ocramius) [reply src/Factory/PdoAdapterFactory.php]: For now, let's just drop it from our local phpcs config (this repo) via exclusion. Should be like https://github.com/Ocramius/ProxyManager/blob/8846623bea8749ded206db3ce701e4e508d516db/phpcs.xml.dist#L20-L21

**Comment by Ocramius** [comment]
Excellent, also thanks for squashing!

**Review by Ocramius — APPROVED** [review]
Thanks @snapshotpl!

**Pull request merged**
